plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbTypeMapping.java

```diff
@@ -0,0 +1,726 @@
+    }
+
+    @Test
+    public void testDouble()
+    {
+        // we are not testing Nan/-Infinity/+Infinity as those are not supported by MariaDB
+        SqlDataTypeTest.create()
+                .addRoundTrip("double", "3.14", DOUBLE, "CAST(3.14 AS DOUBLE)")
+                .addRoundTrip("double", "1.0E100", DOUBLE, "1.0E100")
+                .addRoundTrip("double", "1.23456E12", DOUBLE, "1.23456E12")
+                .addRoundTrip("double", "NULL", DOUBLE, "CAST(NULL AS DOUBLE)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("trino_test_double"))
+                .execute(getQueryRunner(), mariaDbCreateAndInsert("tpch.test_double"));
+    }
+
+    @Test
+    public void testTrinoCreatedParameterizedVarchar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("varchar(10)", "'text_a'", createVarcharType(255), "CAST('text_a' AS VARCHAR(255))")
+                .addRoundTrip("varchar(255)", "'text_b'", createVarcharType(255), "CAST('text_b' AS VARCHAR(255))")
+                .addRoundTrip("varchar(256)", "'text_c'", createVarcharType(65535), "CAST('text_c' AS VARCHAR(65535))")
+                .addRoundTrip("varchar(65535)", "'text_d'", createVarcharType(65535), "CAST('text_d' AS VARCHAR(65535))")
+                .addRoundTrip("varchar(65536)", "'text_e'", createVarcharType(16777215), "CAST('text_e' AS VARCHAR(16777215))")
+                .addRoundTrip("varchar(16777215)", "'text_f'", createVarcharType(16777215), "CAST('text_f' AS VARCHAR(16777215))")
+                .addRoundTrip("varchar(16777216)", "'text_g'", createUnboundedVarcharType(), "CAST('text_g' AS VARCHAR)")
+                .addRoundTrip("varchar(2147483646)", "'text_h'", createUnboundedVarcharType(), "CAST('text_h' AS VARCHAR)")
+                .addRoundTrip("varchar", "'unbounded'", createUnboundedVarcharType(), "CAST('unbounded' AS VARCHAR)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("trino_test_parameterized_varchar"));
+    }
+
+    @Test
+    public void testMariaDbCreatedParameterizedVarchar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("tinytext", "'a'", createVarcharType(255), "CAST('a' AS VARCHAR(255))")
+                .addRoundTrip("text", "'b'", createVarcharType(65535), "CAST('b' AS VARCHAR(65535))")
+                .addRoundTrip("mediumtext", "'c'", createVarcharType(16777215), "CAST('c' AS VARCHAR(16777215))")
+                .addRoundTrip("longtext", "'d'", createUnboundedVarcharType(), "CAST('d' AS VARCHAR)")
+                .addRoundTrip("varchar(32)", "'e'", createVarcharType(32), "CAST('e' AS VARCHAR(32))")
+                .addRoundTrip("varchar(15000)", "'f'", createVarcharType(15000), "CAST('f' AS VARCHAR(15000))")
+                .execute(getQueryRunner(), mariaDbCreateAndInsert("tpch.mariadb_test_parameterized_varchar"));
+    }
+
+    @Test
+    public void testMariaDbCreatedParameterizedVarcharUnicode()
+    {
+        SqlDataTypeTest.create()
```

Comment:
Tests with 4 byte characters are missing. If it's intentional add a test that asserts failure.

Reply:
I guess this is because MariaDB doesn't have the utf8mb4 encoding and utf8 is limited to 3 bytes in MariaDB (same as old MySQL versions).